pkg/drivers/powerflex.go

```diff
@@ -0,0 +1,174 @@
+package drivers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	csmv1 "github.com/dell/csm-operator/api/v1"
+	"github.com/dell/csm-operator/pkg/logger"
+	"github.com/dell/csm-operator/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+// PrecheckPowerFlex do input validation
+func PrecheckPowerFlex(ctx context.Context, cr *csmv1.ContainerStorageModule, operatorConfig utils.OperatorConfig, ct client.Client) error {
+
+	log := logger.GetLogger(ctx)
+
+	// Check if driver version is supported by doing a stat on a config file
+	configFilePath := fmt.Sprintf("%s/driverconfig/%s/%s/upgrade-path.yaml", operatorConfig.ConfigDirectory, csmv1.PowerFlex, cr.Spec.Driver.ConfigVersion)
+	if _, err := os.Stat(configFilePath); os.IsNotExist(err) {
+		log.Errorw("PreCheckPowerFlex failed in version check", "Error", err.Error())
+		return fmt.Errorf("%s %s not supported", csmv1.PowerFlexName, cr.Spec.Driver.ConfigVersion)
+	}
+
+	//driver := cr.GetDriverType()
+	mdmVar, err := GetMDMFromSecret(ctx, cr, ct)
+	if err != nil {
+		return err
+	}
+	var newmdm corev1.EnvVar
+	for _, initcontainer := range cr.Spec.Driver.InitContainers {
+		if initcontainer.Name == "sdc" {
+			k := 0
+			initenv := initcontainer.Envs
+			for c, env := range initenv {
+				if env.Name == "MDM" {
+					env.Value = mdmVar
+					newmdm = env
+					k = c
+					break
+				}
+			}
+			initenv[k] = newmdm
+			break
+		}
+	}
+	return nil
+}
+
+// GetMDMFromSecret - Get MDM value from secret
+func GetMDMFromSecret(ctx context.Context, cr *csmv1.ContainerStorageModule, ct client.Client) (string, error) {
+
+	log := logger.GetLogger(ctx)
+	//secretName := fmt.Sprintf("%s-config", cr.Name)
+	secretName := cr.Name + "-config"
+	credSecret, err := utils.GetSecret(ctx, secretName, cr.GetNamespace(), ct)
+	if err != nil {
+		return "", fmt.Errorf("reading secret [%s] error [%s]", secretName, err)
+	}
+
+	type StorageArrayConfig struct {
```

Comment:
Do we want this here or in the api?

Reply:
in the same file 
its just related to secret and we dont use it anywhere else